pkg/metrics/collector/snapshotter.go

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2022. Nydus Developers. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package collector
+
+import (
+	"fmt"
+	"os/exec"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/containerd/containerd/log"
+	"github.com/containerd/nydus-snapshotter/pkg/metrics/data"
+)
+
+type SnapshotterMetricsCollector struct{}
+type SnapshotterCacheUsageCollector struct{}
+
+var GlobalSnapshotterMetricsCollector *SnapshotterMetricsCollector
+var GlobalSnapshotterCacheUsageCollector *SnapshotterCacheUsageCollector
+
+func init() {
+	GlobalSnapshotterMetricsCollector = &SnapshotterMetricsCollector{}
+	GlobalSnapshotterCacheUsageCollector = &SnapshotterCacheUsageCollector{}
+}
+
+type SnapshotterMethod string
+
+const (
+	SnapshotterMethodUnknown SnapshotterMethod = "UNKNOWN"
+	SnapshotterMethodPrepare SnapshotterMethod = "PREPARE"
+	SnapshotterMethodMount   SnapshotterMethod = "MOUNT"
+	SnapshotterMethodRemove  SnapshotterMethod = "REMOVE"
+)
+
+func (s *SnapshotterMetricsCollector) Collect(snapshotID string, begin, end time.Time, method SnapshotterMethod) {
+	beginTime := begin.Format("2006-01-02 15:04:05.000")
+	endTime := end.Format("2006-01-02 15:04:05.000")
+	elapsed, _ := strconv.ParseFloat(fmt.Sprintf("%.6f", end.Sub(begin).Seconds()*1000), 64)
```

Comment:
Call `Milliseconds` rather than Seconds/1000 ?

Reply:
Yes, replaced it.

Reply:
> Call `Milliseconds` rather than Seconds/1000 ?

Shall we replace it with `Nanoseconds()) / 1e6` to show nanoseconds?